src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -659,7 +659,9 @@ bool BLEManagerImpl::SendWriteRequest(BLE_CONNECTION_OBJECT conId, const ChipBle
         return false;
     }
 
-    rc = ble_gattc_write_flat(conId, chr->chr.val_handle, pBuf->Start(), pBuf->DataLength(), OnWriteComplete, this);
+    assert(pBuf->DataLength() <= UINT16_MAX);
```

Comment:
Maybe you should return an error code here instead of aborting?

Reply:
Done